growthbook/growthbook
# Introduce team management UI

### Features and Changes


Introduce a team management UI within the Team tab. Allows users to create teams within the GrowthBook UI as well as via SCIM (currently only works with Okta.) Any teams created via SCIM will need to have membership and name changes done through Okta rather than the GrowthBook UI.



### Dependencies

#1762 


### Testing


Attempt to do CRUD operations for teams

1. Create a team
2. Add members to the team
3. Modify team metadata
4. Modify team permissions
5. Try to delete a team with members (should error)
6. Delete team members
7. Try to delete a team with no members

Test features being blocked for manageByIdp teams (Name changes, membership changes)

1. Edit a previously created team in the teams collection and set `managedByIdp` to true
2. Ensure that you're unable to edit team name and team membership
3. Ensure that you see a banner explaining what is restricted

### Screenshots


#### Teams tab
![Screenshot 2023-10-26 at 11 00 31 AM](https://github.com/growthbook/growthbook/assets/14947905/dc94d28b-65fe-44d2-892e-314cb8997f6b)

#### Create Team Modal
![Screenshot 2023-10-26 at 11 00 22 AM](https://github.com/growthbook/growthbook/assets/14947905/e89f80fe-6b6a-4c32-adcf-b1ae49e9583c)

#### GrowthBook Managed Team
![Screenshot 2023-10-26 at 11 00 42 AM](https://github.com/growthbook/growthbook/assets/14947905/142ec9ec-39a6-46c8-915f-7568fe68e960)

#### Okta Managed Team
![Screenshot 2023-10-26 at 11 01 18 AM](https://github.com/growthbook/growthbook/assets/14947905/0cc66857-277f-4616-80a3-3b8ebee0dd37)

#### Add Team Members Modal
![Screenshot 2023-10-26 at 11 04 56 AM](https://github.com/growthbook/growthbook/assets/14947905/3873bcc4-c26b-4970-94fe-3291deacfb4d)

#### Teams page with upgrade message
![Screenshot 2023-10-26 at 11 07 01 AM](https://github.com/growthbook/growthbook/assets/14947905/0956dbb2-0a61-4eec-99aa-f83fdad89e94)
